docker-entrypoint.sh

```diff
@@ -1,110 +1,129 @@
+    php yii migrate/up --includeModuleMigrations=1 --interactive=0
+    php yii search/rebuild
+    cp -va /tmp/humhub/.version "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/.version"
+  fi
 else
-	echo "No existing installation found!"
-	echo "Installing source files..."
-	cp -rv /usr/src/humhub/protected/config/* /var/www/localhost/htdocs/protected/config/
-	cp -v /usr/src/humhub/.version /var/www/localhost/htdocs/protected/config/.version
-
-	mkdir -p /var/www/localhost/htdocs/protected/runtime/logs/
-	touch /var/www/localhost/htdocs/protected/runtime/logs/app.log
-
-	echo "Setting permissions..."
-	chown -R nginx:nginx /var/www/localhost/htdocs/uploads
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/modules
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/config
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/runtime
-
-	wait_for_db
-
-	echo "Creating database..."
-	cd /var/www/localhost/htdocs/protected/
-	if [ -z "$HUMHUB_DB_USER" ]; then
-		AUTOINSTALL="false"
-	fi
-
-	if [ "$AUTOINSTALL" != "false" ]; then
-		echo "Installing..."
-		php yii installer/write-db-config "$HUMHUB_DB_HOST" "$HUMHUB_DB_NAME" "$HUMHUB_DB_USER" "$HUMHUB_DB_PASSWORD"
-		php yii installer/install-db
-		php yii installer/write-site-config "$HUMHUB_NAME" "$HUMHUB_EMAIL"
-		php yii installer/create-admin-account
-		chown -R nginx:nginx /var/www/localhost/htdocs/protected/runtime
-	fi
+  echo "No existing installation found!"
+  echo "Installing source files..."
+  if [ ! -d "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}" ]; then
+    echo "Moving humhub files to ${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    mv /tmp/humhub "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+  else
+    mkdir -p "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    echo "Copying files into ${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    cp -a /tmp/humhub/* "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/"
+    find /tmp/humhub/ -regex '.*/\..*' -maxdepth 1 | xargs -I{} cp -a {} /var/www/localhost/htdocs/
+  fi
+  [ -d "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/protected/config/" ] || mkdir "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/protected/config/"
+  cp -av /usr/src/humhub/protected/config/* "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/protected/config/"
+
+  wait_for_db
+
+  echo "Creating database..."
+  cd "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/protected/"
+  if [ -z "$HUMHUB_DB_USER" ]; then
+    AUTOINSTALL="false"
+  fi
+
+  if [ "$AUTOINSTALL" != "false" ]; then
+    echo "Installing..."
+    php yii installer/write-db-config "$HUMHUB_DB_HOST" "$HUMHUB_DB_NAME" "$HUMHUB_DB_USER" "$HUMHUB_DB_PASSWORD"
+    php yii installer/install-db
+    php yii installer/write-site-config "$HUMHUB_NAME" "$HUMHUB_EMAIL"
+
+    # Set baseUrl if provided
+    if [ -n "$HUMHUB_PROTO" ] && [ -n "$HUMHUB_HOST" ]; then
+      HUMHUB_BASE_URL="${HUMHUB_PROTO}://${HUMHUB_HOST}${HUMHUB_SUB_DIR}/"
+      echo "Setting base url to: $HUMHUB_BASE_URL"
+      php yii installer/set-base-url "${HUMHUB_BASE_URL}"
```

Comment:
Setting the base-url (even without the subdir install) could be a really good idea. Thx for bringing that up.